src/main/java/com/vinylteam/vinyl/util/impl/DecksParser.java

```diff
@@ -0,0 +1,193 @@
+                }
+            }
+        }
+        log.info("Got genres from Decks.de: {}", genreLinks.size());
+        return genreLinks;
+    }
+
+    private String getHighResImageLinkFromDocument(Document offerDocument) {
+        Element iframeElement = offerDocument.select("iframe").first();
+        return iframeElement.parents().get(0).getElementsByClass("bigCoverDetail").select("img")
+                .attr("src");
+    }
+
+    String getGenreFromDocument(Document offerDocument) {
+        Element iframeElement = offerDocument.select("iframe").first();
+        return iframeElement.parents().get(0).getElementsByClass("LStylehead").text();
+    }
+
+    Optional<Currency> getOptionalCurrencyFromDocument(Document offerDocument) {
+        Element iframeElement = offerDocument.select("iframe").first();
+        String priceDetails = iframeElement.parents().get(0).getElementsByClass("preisschild").text();
+        if (priceDetails.indexOf(' ') != -1) {
+            String currency = priceDetails.substring(priceDetails.indexOf(' ') + 1, priceDetails.indexOf('*') - 1);
+            return Currency.getCurrency(currency);
+        }
+        return Optional.empty();
+    }
+
+    double getPriceFromDocument(Document offerDocument) {
+        double price = 0.;
+        Element iframeElement = offerDocument.select("iframe").first();
+        String priceDetails = iframeElement.parents().get(0).getElementsByClass("preisschild").text();
+        if (priceDetails.indexOf(' ') != -1) {
+            price = Double.parseDouble(priceDetails.substring(0, priceDetails.indexOf(' ')));
+        }
+        return price;
+    }
+
+    String getArtistFromDocument(Document offerDocument) {
+        Element iframeElement = offerDocument.select("iframe").first();
+        return iframeElement.parents().get(0).getElementsByClass("detail_artist  lightCol mainColBG ")
+                .select("h1").text();
+    }
+
+    String getReleaseFromDocument(Document offerDocument) {
+        Element iframeElement = offerDocument.select("iframe").first();
+        return iframeElement.parents().get(0).getElementsByClass("detail_titel  lightCol mainCol ")
+                .select("h1").text();
+    }
+
+    public HashSet<RawOffer> readRawOffersFromAllOfferLinks(HashSet<String> offerLinks) {
+        HashSet<RawOffer> rawOfferSet = new HashSet<>();
+        for (String offerLink : offerLinks) {
+            RawOffer rawOffer = getRawOfferFromOfferLink(offerLink);
+            if (isValid(rawOffer)) {
+                rawOfferSet.add(rawOffer);
+            } else {
+                log.warn("Can't fill raw offer by offer link, not adding it to set {'rawOffer':{}, 'offerLink':{}}",
+                        rawOffer, offerLink);
+            }
+        }
+        log.info("Got row offers from Decks.de: {}", rawOfferSet.size());
+        return rawOfferSet;
+    }
+
+    boolean isValid(RawOffer rawOffer) {
+        boolean isValid = false;
```

Comment:
what about direct setting of the variable?
        boolean isValid = rawOffer.getPrice() != 0d
                && rawOffer.getCurrency().isPresent()
                && !rawOffer.getRelease().isEmpty()
                && rawOffer.getOfferLink() != null;